rewriter/flatten.cc

```diff
@@ -313,14 +325,30 @@ class FlattenWalk {
         }
         classDef->rhs = addClassDefMethods(ctx, std::move(classDef->rhs), classDef->loc);
         return classDef;
-    };
+    }
+
+    bool shouldMoveSend(ast::Send& send) {
+        if (send.fun == core::Names::sig()) {
+            return true;
+        }
+        if ((send.fun == core::Names::private_() ||
+             send.fun == core::Names::privateClassMethod() ||
+             send.fun == core::Names::protected_() ||
+             send.fun == core::Names::public_()) &&
+            send.args.size() == 1 &&
+            send.recv->isSelfReference()) {
+            return true;
+        }
+        return false;
+    }
```

Comment:
These are the sends that we move: sigs and visibility modifiers. All the modifiers listed here we treat specially in the namer.